File: dotnet/xmsclient/Client/Handler/ConnectionOpenOkMethodHandler.cs
using System;
using jpmorgan.mina.common;
using log4net;
using OpenAMQ.Framing;
using OpenAMQ.XMS.Client.Protocol;
using OpenAMQ.XMS.Client.State;

namespace OpenAMQ.XMS.Client.Handler
{
    public class ConnectionOpenOkMethodHandler : IStateAwareMethodListener
    {
        private static readonly ILog _logger = LogManager.GetLogger(typeof(ConnectionOpenOkMethodHandler));

        private static readonly ConnectionOpenOkMethodHandler _instance = new ConnectionOpenOkMethodHandler();

        public static ConnectionOpenOkMethodHandler GetInstance()
        {
            return _instance;
        }

        private ConnectionOpenOkMethodHandler()
        {
        }

        public void MethodReceived(AMQStateManager stateManager, AMQMethodEvent evt)
        {
            AMQProtocolSession session = evt.ProtocolSession;
            ConnectionOpenOkBody method = (ConnectionOpenOkBody) evt.Method;
            stateManager.ChangeState(AMQState.CONNECTION_OPEN);
        }

    }

}

File: java/src/org/openamq/client/handler/ConnectionOpenOkMethodHandler.java
package org.openamq.client.handler;

import org.apache.log4j.Logger;
import org.openamq.AMQException;
import org.openamq.client.protocol.AMQMethodEvent;
import org.openamq.client.protocol.AMQProtocolSession;
import org.openamq.client.state.AMQState;
import org.openamq.client.state.AMQStateManager;
import org.openamq.client.state.StateAwareMethodListener;
import org.openamq.framing.ConnectionOpenOkBody;

public class ConnectionOpenOkMethodHandler implements StateAwareMethodListener
{

    private static final Logger _logger = Logger.getLogger(ConnectionOpenOkMethodHandler.class);

    private static final ConnectionOpenOkMethodHandler _instance = new ConnectionOpenOkMethodHandler();

    public static ConnectionOpenOkMethodHandler getInstance()
    {
        return _instance;
    }

    private ConnectionOpenOkMethodHandler()
    {
    }

    public void methodReceived(AMQStateManager stateManager, AMQMethodEvent evt) throws AMQException
    {
        AMQProtocolSession session = evt.getProtocolSession();
        ConnectionOpenOkBody method = (ConnectionOpenOkBody) evt.getMethod();        
        stateManager.changeState(AMQState.CONNECTION_OPEN);
    }

}
